runtime/autoload/remote/host.vim

```diff
@@ -118,7 +118,38 @@ function! remote#host#RegisterPlugin(host, path, specs) abort
 endfunction
 
 
-function! s:GetManifest() abort
+" Get the path to the rplugin manifest file.
+function! s:GetManifestPath() abort
+  let manifest_base = ''
+
+  if exists('$NVIM_RPLUGIN_MANIFEST')
+    return fnamemodify($NVIM_RPLUGIN_MANIFEST, ':p')
+  endif
+
+  let preferred = has('win32')
+        \ ? ['$LOCALAPPDATA', '~/AppData/Local']
+        \ : ['$XDG_DATA_HOME', '~/.local/share']
+
+  for dest in preferred
+    if dest[0] == '$' && !exists(dest)
+      continue
+    endif
+
+    let dest = fnamemodify(expand(dest), ':p')
+    if !empty(dest) && isdirectory(dest)
+      let dest .= 'nvim/'
+      call mkdir(dest, 'p', 700)
```

Comment:
the restrictive permissions could be what's failing QB. it's had problems in the past where it couldn't delete files because of restrictive permissions.